syntax = "proto3";

package vikraft;

message Connect {
    int32 id = 1;
}

message RequestVote {
    int32 term = 1;
    int32 candidate_id = 2;
    int32 last_log_index = 3;
    int32 last_log_term = 4;
}

message RequestVoteResponse {
    int32 term = 1;
    bool vote_granted = 2;
}

message AppendEntries {
    int32 term = 1;
    int32 leader_id = 2;
    int32 prev_log_index = 3;
    int32 prev_log_term = 4;
}

message AppendEntriesResponse {
    int32 term = 1;
    bool success = 2;
}
enum MessageType {
    CONNECT = 0;
    REQUEST_VOTE = 1;
    REQUEST_VOTE_RESPONSE = 2;
    APPEND_ENTRIES = 3;
    APPEND_ENTRIES_RESPONSE = 4;
}

message Message {
    MessageType type = 1;
    bytes data = 2;
}